src/main/steps/upload-your-documents/content.ts

```diff
@@ -0,0 +1,198 @@
+    },
+  },
+  submit: {
+    text: l => l.continue,
+  },
+};
+
+const languages = {
+  en,
+  cy,
+};
+
+const getFormWithCustomContent = (content: CommonContent) => {
+  const newForm = cloneDeep(form);
+
+  (newForm.fields.uploadedDocuments as FormInput).value = isObject(content.formState?.uploadedDocuments)
+    ? JSON.stringify(content.formState?.uploadedDocuments || [])
+    : content.formState?.uploadedDocuments || '[]';
+
+  const checkboxes: { id: string; value: SupportingDocumentType }[] = [];
+
+  if (content.formState?.inTheUk === YesOrNo.YES) {
+    checkboxes.push({
+      id: 'cannotUploadCertificate',
+      value: SupportingDocumentType.UNION_CERTIFICATE,
+    });
+  } else {
+    checkboxes.push({
+      id: 'cannotUploadForeignCertificate',
+      value: SupportingDocumentType.FOREIGN_UNION_CERTIFICATE,
+    });
+  }
+
+  if (content.formState?.certifiedTranslation === YesOrNo.YES) {
+    checkboxes.push({
+      id: 'cannotUploadForeignCertificateTranslation',
+      value: SupportingDocumentType.FOREIGN_UNION_CERTIFICATE_TRANSLATION,
+    });
+  }
+
+  if (
+    content.formState?.lastNameChangeWhenRelationshipFormed === YesOrNo.YES ||
+    content.formState?.anyNameChangeSinceRelationshipFormed === YesOrNo.YES
+  ) {
+    checkboxes.push({
+      id: 'cannotUploadNameChangeProof',
+      value: SupportingDocumentType.NAME_CHANGE_PROOF,
+    });
+  }
+
+  if (checkboxes.length > 1) {
+    newForm.fields.cannotUpload = {
+      type: 'checkboxes',
+      label: l => l.cannotUploadDocuments,
+      labelHidden: true,
+      validator: (value, formData) => {
+        if (value && !formData.cannotUploadDocuments?.length) {
+          return 'required';
+        }
+      },
+      values: [
+        {
+          name: 'cannotUpload',
+          label: l => l.cannotUploadDocuments,
+          value: Checkbox.Checked,
+          subFields: {
+            cannotUploadDocuments: {
+              type: 'checkboxes',
+              label: l => l.cannotUploadWhich,
+              hint: l => l.checkAllThatApply,
+              subtext: l => l.cannotUploadYouCanPost,
+              validator: atLeastOneFieldIsChecked,
+              values: checkboxes.map(checkbox => ({
+                name: 'cannotUploadDocuments',
+                label: l => l[checkbox.id],
+                value: checkbox.value,
+              })),
+            },
+          },
+        },
+      ],
+    };
+  } else {
+    newForm.fields.cannotUploadDocuments = {
+      type: 'checkboxes',
+      subtext: l => l.cannotUploadYouCanPost,
+      values: checkboxes.map(checkbox => ({
+        name: 'cannotUploadDocuments',
+        label: l => l[`${checkbox.id}Singular`],
+        value: checkbox.value,
+      })),
+    };
+  }
+
+  return newForm;
+};
+
+export const generateContent: TranslationFn = content => {
+  const translations = languages[content.language](content);
+  return {
+    ...translations,
+    form: getFormWithCustomContent(content),
+  };
+};
```

Comment:
I'm not sure all of that is great, but if you think it's great then that's good enough for me :+1:

Reply:
Yeah, I wasn't happy with cloning the form and modifying it either, so I've refactored this so that `form.fields` can now also be a function which takes the `formState` and then can dynamically change questions based on previous responses.